pkg/apis/pingcap/v1alpha1/types.go

```diff
@@ -1437,3 +1441,272 @@ type IngressSpec struct {
 	// +optional
 	TLS []extensionsv1beta1.IngressTLS `json:"tls,omitempty"`
 }
+
+// +k8s:openapi-gen=true
+// DMCluster is the control script's spec
+type DMCluster struct {
+	metav1.TypeMeta `json:",inline"`
+	// +k8s:openapi-gen=false
+	metav1.ObjectMeta `json:"metadata"`
+
+	// Spec defines the behavior of a dm cluster
+	Spec DMClusterSpec `json:"spec"`
+
+	// +k8s:openapi-gen=false
+	// Most recently observed status of the dm cluster
+	Status DMClusterStatus `json:"status"`
+}
+
+// +k8s:openapi-gen=true
+// DMDiscoverySpec contains details of Discovery members for dm
+type DMDiscoverySpec struct {
+	corev1.ResourceRequirements `json:",inline"`
+	// Host indicates the existed TiDB discovery host. If not specified, dm cluster will start a new one
+	Host string `json:"host"`
+}
+
+// +k8s:openapi-gen=true
+// DMClusterSpec describes the attributes that a user creates on a dm cluster
+type DMClusterSpec struct {
+	// Discovery spec
+	Discovery DMDiscoverySpec `json:"discovery,omitempty"`
+
+	// dm-master cluster spec
+	// +optional
+	Master *MasterSpec `json:"dm_master,omitempty"`
+
+	// dm-worker cluster spec
+	// +optional
+	Worker *WorkerSpec `json:"dm_worker,omitempty"`
+
+	// Indicates that the dm cluster is paused and will not be processed by
+	// the controller.
+	// +optional
+	Paused bool `json:"paused,omitempty"`
+
+	// TODO: remove optional after defaulting logic introduced
+	// dm cluster version
+	// +optional
+	Version string `json:"version"`
+
+	// SchedulerName of DM cluster Pods
+	// +kubebuilder:default=tidb-scheduler
+	SchedulerName string `json:"schedulerName,omitempty"`
+
+	// Persistent volume reclaim policy applied to the PVs that consumed by DM cluster
+	// +kubebuilder:default=Retain
+	PVReclaimPolicy *corev1.PersistentVolumeReclaimPolicy `json:"pvReclaimPolicy,omitempty"`
+
+	// ImagePullPolicy of DM cluster Pods
+	// +kubebuilder:default=IfNotPresent
+	ImagePullPolicy corev1.PullPolicy `json:"imagePullPolicy,omitempty"`
+
+	// ImagePullSecrets is an optional list of references to secrets in the same namespace to use for pulling any of the images.
+	// +optional
+	ImagePullSecrets []corev1.LocalObjectReference `json:"imagePullSecrets,omitempty"`
+
+	// ConfigUpdateStrategy determines how the configuration change is applied to the cluster.
+	// UpdateStrategyInPlace will update the ConfigMap of configuration in-place and an extra rolling-update of the
+	// cluster component is needed to reload the configuration change.
+	// UpdateStrategyRollingUpdate will create a new ConfigMap with the new configuration and rolling-update the
+	// related components to use the new ConfigMap, that is, the new configuration will be applied automatically.
+	// +kubebuilder:validation:Enum=InPlace,RollingUpdate
+	// +kubebuilder:default=InPlacne
```

Comment:
Please make RollingUpdate as default, which is what we recommend for production, and for TC, the InPlace is just for backward compatibility.